### PR TITLE
added instructions for adding the modules.

### PR DESCRIPTION
**Pull request by dhirensr**

### CODE REVIEW

**Comment by Ali-Razmjoo** [comment]
can we have docs also on wiki?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -126,3 +126,80 @@ Finally, go to your repository in browser and click on `compare and pull request
...
+Describing about the module
+- Dockerfile
+For setting up all the packages,libraries,scripts to run by the module.
+
+
+### Testing the module
+
+For testing the module run the command
+```
+python3 ohp.py -m module_name/subpart
```
Comment (ChakshuGupta) [inline CONTRIBUTING.md:204]: with "--test"

**Comment by ChakshuGupta** [comment]
@Ali-Razmjoo We can copy this to wiki, once this is reviewed.

File: CONTRIBUTING.md (hunk 1 above)
Comment (dhirensr) [reply CONTRIBUTING.md:204]: @ChakshuGupta : module testing is with -m to run the module right?

Comment (ChakshuGupta) [reply CONTRIBUTING.md:204]: That would be for running the module and yes you can test it like that but technical testing would be with the `--test` flag.

Comment (Ali-Razmjoo) [inline CONTRIBUTING.md:204]:
```suggestion
python3 ohp.py -m protocol/moduleOrType
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -126,3 +126,80 @@ Finally, go to your repository in browser and click on `compare and pull request
 Add a title and description to your pull request that explains your precious effort.
 
 Sit and relax till we review your PR, you've made your contribution to our project.
+
+
+## Adding a Module
+
+OWASP Python Honeypot currently supports 7 modules namely FTP - strong and weak password, SSH - strong and weak password , HTTP - basic auth strong passsword and basic auth weak password and ICS honeypot module.
```
Comment (Ali-Razmjoo) [inline CONTRIBUTING.md:133]:
```suggestion
OWASP Python Honeypot currently supports multiple types of protocols with different types and modules for various purposes like getting credentials, network events, files, honeypot events, and custom data coming from each module.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -126,3 +126,80 @@ Finally, go to your repository in browser and click on `compare and pull request
...
+
+### To add a new module on can create a new folder inside the modules directory of the project.
```
Comment (Ali-Razmjoo) [inline CONTRIBUTING.md:135]: it's not clear to understand.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -126,3 +126,80 @@ Finally, go to your repository in browser and click on `compare and pull request
...
+
+### Set up module files
+Each module has an init.py file  which has the module category configuration, below shown is the template for the same.
```
Comment (Ali-Razmjoo) [inline CONTRIBUTING.md:138]: `__init__.py`?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -126,3 +126,80 @@ Finally, go to your repository in browser and click on `compare and pull request
...
+        "virtual_machine_internet_access": Bool,
+        "real_machine_port_number": DOCKER HOST PORT NUMBER
+    }
+```
+
+Then if the module has sub parts like weak and strong password then two separate folders should be created.
+Inside the module folder there should be:
+- files folder
+if the modules require some extra scripts/config files which needs to be moved to the module containers.
+- init.py
```
Comment (Ali-Razmjoo) [inline CONTRIBUTING.md:158]: `__init__.py`

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -126,3 +126,80 @@ Finally, go to your repository in browser and click on `compare and pull request
...
+        "virtual_machine_port_number": PORT NUMBER,
+        "real_machine_port_number": PORT NUMBER FOR DOCKER HOST,
+        "extra_docker_options": [""],
+        "module_processor": ModuleProcessor()
+    }
+```
+- readme.md
+Describing about the module
+- Dockerfile
+For setting up all the packages,libraries,scripts to run by the module.
```
Comment (Ali-Razmjoo) [inline CONTRIBUTING.md:197]:
```suggestion
For setting up all the packages, libraries, scripts to run by the module.
```

**Comment by dhirensr** [comment]
> add database and events section, description, and purposes.

Should it be the same content from (https://github.com/zdresearch/OWASP-Honeypot/wiki/Database) in this file too?

**Comment by Ali-Razmjoo** [comment]
@dhirensr yes, it would bst to let the developer know how to use it and they exist

**Comment by Ali-Razmjoo** [comment]
let's use this PR also for updating wiki, 

1- https://github.com/zdresearch/OWASP-Honeypot/wiki/Installation, on this page,  we need to add information about client/server, and how to install the stuff it needed for each. maybe by starting to explain what's different between server, client and how to run each, also we need to mention using docker in here.

2- https://github.com/zdresearch/OWASP-Honeypot/wiki/Modules, on this page we need to mention these existing modules for each protocol it's just a sample for the framework, by studying them we can develop more of our own, and ofcourse we can use them as how they are now., and maybe rename the page to "available modules"?

3- https://github.com/zdresearch/OWASP-Honeypot/wiki/Usage, on this page we need to seperate developer docs and it must be only for command-line sample and usages, and maybe create another page for "developer guidelines" to create their own modules, use the current features in the framework (such as database events and alert or logs or the features we have for modules.

4- https://github.com/zdresearch/OWASP-Honeypot/wiki/API, this need to be updated with new api (/explore section, responses are not updated)

5- https://github.com/zdresearch/OWASP-Honeypot/wiki/Database, add database related functions (introdution and usage maybe?) how to use them in a module in here.

6- https://github.com/zdresearch/OWASP-Honeypot/wiki/How-to-Contribute, maybe add the developer guideline under this one?

and I just updated the event page myself, feel free to modify it with more details. https://github.com/zdresearch/OWASP-Honeypot/wiki/Events

**Comment by Ali-Razmjoo** [comment]
also let's handle #153 in this PR.

**Comment by dhirensr** [comment]
@Ali-Razmjoo : all responses should be updated in https://github.com/zdresearch/OWASP-Honeypot/wiki/API , could you tell which one is differing for you?

**Comment by ChakshuGupta** [comment]
@dhirensr The response for "GET /api/events/explore/<event_type>" isn't updated. Also  "GET /api/pcap/explore" is no longer there. It is merged with the "events/explore" end point.

**Comment by dhirensr** [comment]
@ChakshuGupta : thank you for the reply. I'll work on points 2,4 and 6 for now.

**Review by Ali-Razmjoo — APPROVED** [review]

**Pull request merged**